.github/workflows/check.yaml

```diff
@@ -0,0 +1,60 @@
+name: check
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+
+permissions:
+  contents: write
+
+env:
+  GOMAXPROCS: 4
+
+jobs:
+  build-go:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Init Go Environment
+        uses: ./.github/actions/init_go
+
+      - name: Build
+        run: make build-go
```

Comment:
Could you also add a similar Makefile, .golangci.yaml and docker-compose than in carrotmq ? They are used in this workflow.

We will not need the `rabbitmq` config in docker-compose from carrotmq, we will only need the linter config.
You could also remove the `up` and `down` command in the Makefile, they are not really relevant for this library.